### DISCUSSION TITLE
Image Editing Experience

### DISCUSSION THREAD

**Discussion by LukasV3** [craftcms/cms#10636] [General]
We as an agency are currently looking to improve the overall image editing experience for our clients. The crux of our issue stems from the following scenario;

- An image is given a recommended size using the CMS helper text but is then inevitably resized and cropped throughout the site, looking vastly different to what the client had originally intended. The image may look great as a full-width hero, but on a small preview card, useful information contained within the image can no longer be seen.

We originally tried being more descriptive within the helper text to ensure clients were uploading images at the correct size. However, we still couldn't guarantee that the image wouldn't be cropped and therefore look considerably different on the frontend.

Would love to hear any suggestions on how others have approached this issue and if there are features native to Craft that could be useful here. We have thought about potentially using aspect ratios instead of image dimensions for helper text, which seems promising as image aspect ratios can be edited directly within the CMS. 

Any ideas would be greatly appreciated!

**Comment by wsydney76** [comment]
A few key points from working on an image-heavy site:

All of this is a bit overkill when pictures are used only as 'decoration', but perhaps you find something useful.

* Use live preview a lot
* Use setting a focus point a lot
* Add additional preview targets where the images can be seen in a different context (e.g homepage, archive pages)
* A preview target does not have to be a 'real' frontend page, you can add a dummy page that shows your hero image in all other relevant sizes
* This is especially helpfull if you want to see the effects of the focus point/cropping/resizing on all relevant image sizes at a glance, including responsive sizes. You can also add some instructions (You don't like what you see? Do the following...)
* Consider adding a additional (optional) image versions alongside the original for special purpuses like your preview cards.
* This can be the same image, just with a different focus point, or a slightly different image to prevent the original beeing scaled in a weird way. Or a completely different image, as clicking on a preview card and than seeing the same image again, only bigger, can be a bit boring... Or if the original doesn't play well with the other images on an archive page. Or if you want to use different images for different view port orientations. Or...
* If no second image is assigned, the original version will be used
* You can add that second image to the entry types field layout, or, more reusable, to the asset field layout.
* Take care of image quality. Hello #10371 
* It's always best to prepare images for the web before uploading them (resizing to the biggest version, color/contrast retoning, reducing file size etc). You can than check the size of the biggest version, and if it matches, serve the original without applying any craft transforms. Smaller sizes can than be transformed a high quality. (To be checked, the transform algorithms sometimes produce smaller images with bigger file size)
* If your images are not prepared like this, use a lower quality, and optionally add required effects like sharpening (imager-x plugin required)

**Reply by LukasV3** [reply]
Thanks for this @wsydney76! Lots of useful info here. The addition of preview targets is something we haven't considered and will certainly be looking into!
